activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -370,8 +370,8 @@ def lock_thread=(lock_thread)
       #
       # #connection can be called any number of times; the connection is
       # held in a cache keyed by a thread.
-      def connection
-        @thread_cached_conns[connection_cache_key(@lock_thread || Thread.current)] ||= checkout
+      def connection(owner_thread = @lock_thread || Thread.current)
```

Comment:
I don't like this.. it promotes what's supposed to be a hidden "jam everything, just trust me" switch to being a central component of the class's main public API.